### ISSUE TITLE
[a11y]: ActionableNotification - Screen Reader focus is not trapped within the alertdialog

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#14691]
[Original report](https://ibm-studios.slack.com/archives/C054WK4ES5S/p1694709381258889)

> Issue:
> 1. When the dialog is displayed initial focus is not set with in a dialog.
> 2. In TAB and Shift TAB navigation Screen Reader focus is not trapped with in a dialog
> 
> Recommended fix: 
> 1. When a dialog is displayed the Screen Reader focus needs to be set with in adialog and also the focus needs to be trapped with in a dialog while TAB and Shift TAB navigation and arrow key navigation.
>
> User Impact:
> 
> 1. When the screen reader focus is not restricted within the modal dialog, it also identifies parent page content that disorients screen reader users.
When the focus is not set on a logical point in the page after a user action, it creates a great deal of confusion. The whole process of navigating back to the relevant element can be tedious. note this defect mainly impacts keyboard-only users and those with cognitive disabilities. 
>
> Relates to WCAG 2.1 - 2.4.3 Focus Order 

It's also unclear if the `hasFocus` prop is working and pulling focus to the ActionableNotification on initial render. This was tested in JAWS FF and Voiceover safari iPhone but is present in all browsers.
